pkg/capabilities/capabilities.go

```diff
@@ -370,6 +370,8 @@ outerLoop:
 			}
 
 			vs = append(vs, response.Value)
+		case <-parentCtx.Done():
+			return nil, fmt.Errorf("parent context cancelled: %w", parentCtx.Err())
```

Comment:
actually would be better to return 'capabilities.ErrStopExecution' here I think, @cedric-cordenier agree?